sympy/matrices/sparse.py

```diff
@@ -985,8 +985,10 @@ def col_join(self, other):
         >>> C == A.row_insert(A.rows, Matrix(B))
         True
         """
-        if not self:
-            return type(self)(other)
+        # A null matrix can always be stacked (see  #10770)
+        if self.rows == 0 and self.cols != other.cols:
```

Comment:
It would be possible.  I guess I just have an aesthetic preference for recursion.  If the logic about type casting changes later, the current solution will require no modification, whereas returning the correctly sized matrix directly would require the typecasting logic to be in two places.